### PR TITLE
feat: profiling/flamegraph

### PR DESCRIPTION
**Pull request by ChaoticTempest**
With this we can generate flamegraphs via:

```sh
cargo flamegraph --root --profile flamegraph --test lib -- test_basic_action
```

note that we need to install the `cargo-flamegraph` binary first

Also, no need to manually run `cargo build --release` before each test anymore. `cargo test` will now pick it up

### CODE REVIEW

File: integration-tests/Cargo.toml [hunk 1]
```diff
@@ -44,3 +45,4 @@ reqwest = "0.11.16"
 [features]
 default = []
 docker-test = []
+flamegraph = ["mpc-recovery/disable-open-telemetry"]
```
Comment (volovyks) [inline integration-tests/Cargo.toml:48]: Why do we need to do this?

File: integration-tests/src/env/local.rs [hunk 2]
```diff
@@ -19,8 +19,7 @@ pub struct SignerNode {
     gcp_datastore_url: String,
 
     // process held so it's not dropped. Once dropped, process will be killed.
-    #[allow(unused)]
-    process: Child,
+    _process: NodeProcess,
```
Comment (volovyks) [inline integration-tests/src/env/local.rs:22]: Why are we renaming this?

**Review by volovyks — APPROVED** [review]
Why is it asking for a password?
I failed to run them. My setup is probably different from yours (I'm using nix).

File: integration-tests/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChaoticTempest) [reply integration-tests/Cargo.toml:48]: open telemetry logging messes things up on threaded mode, so it has to be disabled during flamegraph generation

File: integration-tests/src/env/local.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChaoticTempest) [reply integration-tests/src/env/local.rs:22]: better practice than using an `allow(unused)`

File: integration-tests/src/mpc.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
+use std::path::{Path, PathBuf};
```
[the remark this answers is not in the document]
Comment (ChaoticTempest) [reply integration-tests/src/mpc.rs:1]: this is all just logic relating to spawning mpc, so it was better to have it be separate as it is the whole build system we have for it, which shouldn't be mixed with anything else

**Comment by ChaoticTempest** [comment]
> Why is it asking for a password?

It's required to run `dtrace` which monitors the whole system. You can use `sudo` instead of having `--root`

> I failed to run them. My setup is probably different from yours (I'm using nix).

what's the error message you're getting? Shouldn't require anything extra besides that cargo-flamegraph binary.

**Comment by volovyks** [comment]
@ChaoticTempest its:
```
= note: ld: framework not found Security
          clang-11: error: linker command failed with exit code 1 (use -v to see invocation)


error: could not compile `multi-party-eddsa` (lib) due to previous error
```

I was it before, maybe something on my side.

**Review by volovyks — APPROVED** [review]

**Comment by ChaoticTempest** [comment]
@volovyks try this fix: https://www.reddit.com/r/rust/comments/ptagz8/cargo_build_fails_ld_framework_not_found_security/

**Review by itegulov — APPROVED** [review]

**Pull request merged**